File: SPUIKitCategory/UIView/UIView+XIBCreation.m
//
//  UIView+XIBCreation.m
//  SmartCity
//
//  Created by LSP on 2017/5/12.
//  Copyright © 2017年 LSP. All rights reserved.
//

#import "UIView+XIBCreation.h"

@implementation UIView (XIBCreation)

+ (instancetype)loadFromNib {
    return [self loadFromNibWithName:NSStringFromClass([self class])];
}
+ (instancetype)loadFromNibWithName:(NSString *)nibName {
    return [self loadFromNibWithName:nibName owner:nil];
}

+ (instancetype)loadFromNibWithName:(NSString *)nibName owner:(id)owner {
    return [self loadFromNibWithName:nibName owner:owner bundle:[NSBundle mainBundle]];
}

+ (instancetype)loadFromNibWithName:(NSString *)nibName owner:(id)owner bundle:(NSBundle *)bundle {
    NSArray* array = [bundle loadNibNamed:nibName owner:owner options:nil];
    for (id obj in array) {
        if ([obj isKindOfClass:[self class]]) {
            return obj;
        }
    }
    return nil;
}

@end

File: Sources/UIView/UIView+XIBCreation.swift
//
//  UIView+XIBCreation.swift
//  SPUIKitCategory module
//
//  Created by LSP on 2020/12/15.
//  Copyright © 2020 LSP. All rights reserved.
//

#if !os(macOS)
import UIKit

extension UIView {
    
    /// 从XIB文件（与类名相同的xib）初始化。
    /// - Parameters:
    ///   - nibName: 默认与类名相同
    ///   - owner: owner
    ///   - bundle: mainBundle
    /// - Returns: 实例
    public class
    func loadFromNib(_ nibName: String = "", owner: Any? = nil, bundle: Bundle = Bundle.main) -> Self? {
        let name = nibName.isEmpty ? String(describing: self) : nibName
        if let arr = bundle.loadNibNamed(name, owner: owner, options: nil) {
            for obj in arr {
                if let cls = obj as? NSObject, cls is Self  {
                    return cls as? Self
                }
            }
        }
        return nil
    }
}
#endif
